src/main/java/de/themorpheus/edu/taskservice/controller/TaskController.java

```diff
@@ -135,4 +140,17 @@
 		return optionalTask.map(ControllerResult::of).orElseGet(() -> ControllerResult.of(Error.NOT_FOUND, NAME_KEY));
 	}
 
+	public ControllerResult<GetMarkTaskAsDoneResponseDTO> markTaskAsDone(int taskId, Date date) {
+		if (!this.taskRepository.existsById(taskId)) return ControllerResult.of(Error.NOT_FOUND, NAME_KEY);
+
+		if (taskDoneRepository.existsByTaskId(this.taskRepository.getTaskByTaskId(taskId).get()))
```

Comment:
Use repository to get make query isPresent and then store into variable to avoid redundancy and unneccessary queries